src/crewai/utilities/converter.py

```diff
@@ -26,17 +26,24 @@ def to_pydantic(self, current_attempt=1):
             if self.llm.supports_function_calling():
                 return self._create_instructor().to_pydantic()
             else:
-                return self.llm.call(
+                response = self.llm.call(
                     [
                         {"role": "system", "content": self.instructions},
                         {"role": "user", "content": self.text},
                     ]
                 )
+                return self.model.model_validate_json(response)
+        except ValidationError as e:
+            if current_attempt < self.max_attempts:
+                return self.to_pydantic(current_attempt + 1)
+            raise ConverterError(
+                f"Failed to convert text into a Pydantic model due to the following validation error: {e}"
```

Comment:
getting this:
<img width="1239" alt="Screenshot 2025-01-15 at 3 23 40 PM" src="https://github.com/user-attachments/assets/50a97536-6e6e-4902-962b-b5e1e98438a2" />

```python
Operator "<" not supported for types "int" and "int | None"
  Operator "<" not supported for types "int" and "None"
```